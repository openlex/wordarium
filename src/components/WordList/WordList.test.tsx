import React from "react";
import { render, shallow, ShallowWrapper } from "enzyme";
import { WordList, WordListItem } from "@components";
import { wordListMock } from "@mocks/wordList.mock";

describe("WordList markup", () => {
	let wordList: ShallowWrapper = {} as ShallowWrapper;
	const onClick = jest.fn();

	beforeEach(() => {
		wordList = shallow(
			<WordList onClick={onClick} wordList={wordListMock} />
		);
	});

	it("Has same number of element as in array", () => {
		expect(wordList.find(WordListItem).length).toEqual(wordListMock.length);
	});

	it("match snapshoot", () => {
		expect(
			render(<WordList onClick={onClick} wordList={wordListMock} />)
		).toMatchSnapshot();
	});
});
