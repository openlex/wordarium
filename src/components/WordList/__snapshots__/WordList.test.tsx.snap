// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList markup match snapshoot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "data-test-id": "item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-py26lm-WordListItem",
              "data-is-active": "false",
              "data-test-id": "value",
            },
            "children": Array [
              Object {
                "data": "Text",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "data-test-id": "isDifficult",
                },
                "children": Array [
                  Object {
                    "data": "*",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
            Object {
              "attribs": Object {
                "data-test-id": "isDifficult",
              },
              "children": Array [
                Object {
                  "data": "*",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "data-test-id": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text3",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "data-test-id": "item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "css-py26lm-WordListItem",
                    "data-is-active": "false",
                    "data-test-id": "value",
                  },
                  "children": Array [
                    Object {
                      "data": "Text4",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "data-test-id": "item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-py26lm-WordListItem",
              "data-is-active": "false",
              "data-test-id": "value",
            },
            "children": Array [
              Object {
                "data": "Text2",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "data-test-id": "isDifficult",
              },
              "children": Array [
                Object {
                  "data": "*",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
          },
          Object {
            "attribs": Object {
              "data-test-id": "isDifficult",
            },
            "children": Array [
              Object {
                "data": "*",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text3",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "data-test-id": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text4",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "data-test-id": "item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-py26lm-WordListItem",
              "data-is-active": "false",
              "data-test-id": "value",
            },
            "children": Array [
              Object {
                "data": "Text3",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text4",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "data-test-id": "isDifficult",
                },
                "children": Array [
                  Object {
                    "data": "*",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
            Object {
              "attribs": Object {
                "data-test-id": "isDifficult",
              },
              "children": Array [
                Object {
                  "data": "*",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "data-test-id": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "data-test-id": "item",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-py26lm-WordListItem",
              "data-is-active": "false",
              "data-test-id": "value",
            },
            "children": Array [
              Object {
                "data": "Text4",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-is-active": undefined,
              "data-test-id": undefined,
            },
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "data-test-id": "item",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-py26lm-WordListItem",
                "data-is-active": "false",
                "data-test-id": "value",
              },
              "children": Array [
                Object {
                  "data": "Text3",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-is-active": undefined,
                "data-test-id": undefined,
              },
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "data-test-id": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-py26lm-WordListItem",
                  "data-is-active": "false",
                  "data-test-id": "value",
                },
                "children": Array [
                  Object {
                    "data": "Text2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "data-test-id": "isDifficult",
                  },
                  "children": Array [
                    Object {
                      "data": "*",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "data-test-id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "data-test-id": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-is-active": undefined,
                  "data-test-id": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "data-test-id": "isDifficult",
                },
                "children": Array [
                  Object {
                    "data": "*",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "css-py26lm-WordListItem",
                    "data-is-active": "false",
                    "data-test-id": "value",
                  },
                  "children": Array [
                    Object {
                      "data": "Text2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "data-test-id": "item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "css-py26lm-WordListItem",
                    "data-is-active": "false",
                    "data-test-id": "value",
                  },
                  "children": Array [
                    Object {
                      "data": "Text",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-is-active": undefined,
                    "data-test-id": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "data-test-id": undefined,
        },
      },
    ],
    "name": "ul",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
